zilliqa/src/api/zil.rs

```diff
@@ -162,6 +171,12 @@ where
     let s = String::deserialize(deserializer)?;
     s.parse().map_err(serde::de::Error::custom)
 }
+/// Helper function to extract signer address from a public key
+fn extract_signer_address(key: &schnorr::PublicKey) -> Address {
```

Comment:
We already do this work when we call `txn.verify()` in `Consensus::handle_new_transaction`. Can we instead pass the signer up from there?

Reply:
Yes, I considered that, but this code handles both Scilla and EVM transactions. I have two questions:

How can we differentiate between an EVM payment transaction and a smart contract transaction?
How can we determine the contract address for an EVM transaction?